File: twitter-rt/python/rt.py
import time
from collections import defaultdict

t0 = time.time() 

fnlist = []
for line in open("../filename.txt","r"):
    fnlist.append("data/2013-12/{}".format(line.strip()))
for line in open("../filename2.txt","r"):
    fnlist.append("data/2014-01/{}".format(line.strip()))

for i in range(1,2):
    t1 = time.time()
    idset = set()
    for line in open("../../id_list_{}.txt".format(i)):
        idset.add(line.strip())
        
    rt = defaultdict(list)
    for fn in fnlist:
        t2 = time.time()
        print(fn,end="")
        for line in open("../{}".format(fn)):
            sp = line.split("\t")
            if(sp[1] in idset and sp[5] in idset):
                rt[sp[5]].append(sp[1])
        print(" time:{}".format(time.time() - t2))

    f = open("result/rt_result_{}.txt".format(i),"w")
    for k,v in rt.items():
        rt[k] = len(set(v))
    for k,v in sorted(rt.items(),key=lambda x:x[1],reverse=True):
        f.write("{} {}\n".format(k,v))
    f.close
    print(" time_{}:{}".format(i,time.time() - t1))
    
print(" total time:{}".format(time.time() - t0))
   

File: twitter-rt/rt.rb
fnlist = []
open("filename.txt").each do |fn|
  fnlist << fn.rstrip
end
idlist = Hash.new
open("../id_list.txt").each do |id|
  idlist[id.rstrip] = 1
end


rt = Hash.new{|hash,key| hash[key] = Array.new}

fnlist.each do |fn|
  t0 = Time.now
  print fn+" "
  open("data/2014-09/#{fn}").each do |line|
    sp = line.strip.split("\t")
    if idlist[sp[1]] == 1 && idlist[sp[5]] == 1
      rt[sp[5]] << sp[1]
    end
  end
  puts "time:#{Time.now - t0}"
end


open("rt_result.txt","w") do |f|
  rt.each do |k,v|
    
    f.puts "#{k} #{v.uniq.length}"
  end
end
